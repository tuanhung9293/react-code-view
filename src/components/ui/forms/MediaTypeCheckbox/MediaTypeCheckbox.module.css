.wrapper{
    margin: 5px 0;
    overflow: auto;
}

.labels{
    display: flex;
    justify-content: space-between;
    flex-wrap: wrap;
    margin: 5px 0;
}

.inlineFlex {
    display: inline-flex;
    margin: 10px 0;
    height: 40px;
    margin-left: -15px;
}

.checkbox {
    display: flex;
    align-items: center;
    justify-content: space-around;
  }
  
.checkbox > * {
    margin-left: 10px;
    margin-right: 10px;
}
