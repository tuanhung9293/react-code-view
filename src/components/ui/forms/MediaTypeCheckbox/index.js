
import React from 'react'
import Checkbox from 'components/ui/forms/Checkbox'
import Label from 'components/ui/forms/Label'
import styles from './MediaTypeCheckbox.module.css'

function MediaTypeCheckbox({video, image, audio, favorite, onImageCheck, onVideoCheck, onAudioCheck, onFavoriteCheck, label}) {
  return (
    <div className={styles.wrapper}>
      {label &&
        <div className={styles.labels}>
          <Label>{label}</Label>
        </div>
      }
      <div className={styles.inlineFlex}>
        <label className={styles.checkbox}>
          <Checkbox checked={image} onChange={onImageCheck} />Image
        </label>
        <label className={styles.checkbox}>
          <Checkbox checked={video} onChange={onVideoCheck} />Video
        </label>
        <label className={styles.checkbox}>
          <Checkbox checked={audio} onChange={onAudioCheck} />Audio
        </label>
        {favorite !== undefined &&
          <label className={styles.checkbox}>
            <Checkbox checked={favorite} onChange={onFavoriteCheck} />Favorite
          </label>
        }
      </div>
    </div>
  )
}

export default MediaTypeCheckbox