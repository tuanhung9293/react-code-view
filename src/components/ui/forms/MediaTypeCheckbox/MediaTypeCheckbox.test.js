import React from 'react'
import {shallow} from 'enzyme'
import MediaTypeCheckbox from './'

const baseProps = {
  video: true,
  image: true,
  audio: true,
  onImageCheck: () => {},
  onVideoCheck: () => {},
  onAudioCheck: () => {}
}

describe('renders', () => {
  it('renders MediaTypeCheckbox', () => {
    expect(shallow(
        <MediaTypeCheckbox {...baseProps}/>
    )).toMatchSnapshot()
  })

  it('renders MediaTypeCheckbox with label', () => {
    expect(shallow(
        <MediaTypeCheckbox {...baseProps} label="label" />
    )).toMatchSnapshot()
  })

  it('renders MediaTypeCheckbox with favorite checkbox', () => {
    expect(shallow(
        <MediaTypeCheckbox {...baseProps}
          favorite={true}
          onFavoriteCheck={() => {}} />
    )).toMatchSnapshot()
  })
})