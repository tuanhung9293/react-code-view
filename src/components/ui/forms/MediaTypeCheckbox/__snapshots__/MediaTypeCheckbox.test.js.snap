// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders renders MediaTypeCheckbox 1`] = `
<div
  className="wrapper"
>
  <div
    className="inlineFlex"
  >
    <label
      className="checkbox"
    >
      <Checkbox
        checked={true}
        onChange={[Function]}
      />
      Image
    </label>
    <label
      className="checkbox"
    >
      <Checkbox
        checked={true}
        onChange={[Function]}
      />
      Video
    </label>
    <label
      className="checkbox"
    >
      <Checkbox
        checked={true}
        onChange={[Function]}
      />
      Audio
    </label>
  </div>
</div>
`;

exports[`renders renders MediaTypeCheckbox with favorite checkbox 1`] = `
<div
  className="wrapper"
>
  <div
    className="inlineFlex"
  >
    <label
      className="checkbox"
    >
      <Checkbox
        checked={true}
        onChange={[Function]}
      />
      Image
    </label>
    <label
      className="checkbox"
    >
      <Checkbox
        checked={true}
        onChange={[Function]}
      />
      Video
    </label>
    <label
      className="checkbox"
    >
      <Checkbox
        checked={true}
        onChange={[Function]}
      />
      Audio
    </label>
    <label
      className="checkbox"
    >
      <Checkbox
        checked={true}
        onChange={[Function]}
      />
      Favorite
    </label>
  </div>
</div>
`;

exports[`renders renders MediaTypeCheckbox with label 1`] = `
<div
  className="wrapper"
>
  <div
    className="labels"
  >
    <Label>
      label
    </Label>
  </div>
  <div
    className="inlineFlex"
  >
    <label
      className="checkbox"
    >
      <Checkbox
        checked={true}
        onChange={[Function]}
      />
      Image
    </label>
    <label
      className="checkbox"
    >
      <Checkbox
        checked={true}
        onChange={[Function]}
      />
      Video
    </label>
    <label
      className="checkbox"
    >
      <Checkbox
        checked={true}
        onChange={[Function]}
      />
      Audio
    </label>
  </div>
</div>
`;
